### DISCUSSION TITLE
How simultaneous press works

### DISCUSSION THREAD

**Discussion by alf45tar** [alf45tar/PedalinoMini#387] [General]
Simultaneous pressing is not exactly simultaneous because it is quite impossible by a human to be so precise but the implementation take care of it.

Two buttons are considered simultaneous pressed if the second is pressed within 50 ms (PED_SIMULTANEOUS_GAP) from the first. The actions linked to single press of first button are delayed up to 50 ms waiting the second button press. Singles press actions of both buttons never happen if they are pressed with maximum 50 ms of difference. Of course if you are not synchronous enough single press actions are triggered. Increasing PED_SIMULTANEOUS_GAP give user more time but it delays the actions on single press.

Actions delay do not happen if the button is not involved in any simultaneous control.

**Comment by potul** [comment]
Hi, are you leveraging this for simultaneous press detection?

https://github.com/bxparks/AceButton/discussions/83

just asking, as I don't have the sponsor version.

Mat

**Comment by alf45tar** [comment]
No, I am not. I studied the example but it is not applicable to my needs.
